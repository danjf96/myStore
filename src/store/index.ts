
import { applyMiddleware, combineReducers, createStore } from 'redux'
import products from './ducks/products'
import shoppingCart from './ducks/shoppingCart'
import ReduxThunk from 'redux-thunk';

const reducers = {
    products,
    shoppingCart
}

const combinedReducers =  combineReducers(reducers)

const store = createStore(combinedReducers, {}, applyMiddleware(ReduxThunk));

// We can use RootState type in every file in project
declare global {
    type RootState = ReturnType<typeof store.getState>
}

// Thanks to that you will have ability to use useSelector hook with state value
declare module 'react-redux' {
    interface DefaultRootState extends RootState { }
}

export default store

