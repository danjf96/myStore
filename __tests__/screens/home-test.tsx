/**
 * @format
 */

import 'react-native';
import React from 'react';
import { act, fireEvent, render, waitFor } from 'react-native-testing-library';
import Home from '../../src/screens/home';
import store from '../../src/store'
import { Provider } from 'react-redux';

const component = (
  <Provider store={store} > <Home navigation={{ navigate: (screen:'string') => null }}/> </Provider>
);
const fakeTime = (time: number) => new Promise( (resolve, reject) => setTimeout( () => resolve(null) ,time))

it('renders correctly', async () => {
  const screen = render(component)
  expect(screen.toJSON()).toMatchSnapshot()
});

describe('test loading and render items or not', () => {
  const storeProducts = store.getState().products

  test('renders correctly loading', async () => {

    const { getByTestId, getAllByTestId } = render( component )   
    const list = store.getState().products.listProducts
    expect(list).toHaveLength(0)
    expect(getByTestId('loading')).toBeTruthy()

  });

  test('renders correctly card products', async () => {
    const onPress = jest.fn()
    const onPressSecond = jest.fn()

    const { getAllByTestId, getByTestId, getByText, update } = await waitFor(() =>
        render(component)
    ); 

    await waitFor( () => {
      const list = store.getState().products.listProducts
      expect(list).not.toHaveLength(0)
      expect(getAllByTestId('product')).not.toHaveLength(0)

      fireEvent(getAllByTestId('product')[0], 'press')
      expect(getByTestId('InfoModal').props.visible).toBe(true)

      fireEvent(getByText('CONTINUAR COMPRA'), 'press')
      expect(getByTestId('InfoModal').props.visible).toBe(false)

      fireEvent(getAllByTestId('product')[0], 'press')
      fireEvent(getByText('CARRINHO'), 'press')
      const shoppingCart = store.getState().shoppingCart
      expect(shoppingCart.cart).not.toHaveLength(0)

    })

  });
  
}) 
  