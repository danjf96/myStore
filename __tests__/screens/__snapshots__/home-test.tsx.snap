// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  " ",
  <RCTSafeAreaView
    emulateUnlessSupported={true}
  >
    <View
      style={
        Object {
          "padding": 15,
          "width": "100%",
        }
      }
    >
      <RCTScrollView
        collapsable={false}
        data={Array []}
        getItem={[Function]}
        getItemCount={[Function]}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        refreshControl={
          <RefreshControlMock
            onRefresh={[Function]}
            refreshing={false}
          />
        }
        removeClippedSubviews={false}
        renderItem={[Function]}
        renderScrollComponent={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        viewabilityConfigCallbackPairs={Array []}
      >
        <RCTRefreshControl />
        <View />
      </RCTScrollView>
      <ActivityIndicator
        size="large"
        testID="loading"
      />
      <Modal
        animationType="fade"
        hardwareAccelerated={false}
        onPress={[Function]}
        onPressSecondButton={[Function]}
        testID="InfoModal"
        text="Deseja continuar comprando ou ir para o carrinho?"
        textButton="CARRINHO"
        textSecondButton="CONTINUAR COMPRA"
        title="Atenção!"
        transparent={true}
        visible={false}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#353535",
                "flexDirection": "column",
                "height": "100%",
                "justifyContent": "center",
                "opacity": 0.8,
                "position": "absolute",
                "width": "100%",
              }
            }
          />
          <View
            style={
              Object {
                "paddingHorizontal": 40,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 10,
                  "padding": 24,
                  "paddingTop": 32,
                  "width": "100%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "black",
                    "fontSize": 20,
                    "fontWeight": "bold",
                    "lineHeight": 19.6,
                    "marginBottom": 24,
                  }
                }
              >
                Atenção!
              </Text>
              <Text
                style={
                  Object {
                    "color": "black",
                    "fontSize": 16,
                    "letterSpacing": 0.15,
                    "lineHeight": 24,
                    "marginBottom": 24,
                    "textAlign": "left",
                  }
                }
              >
                Deseja continuar comprando ou ir para o carrinho?
              </Text>
              <View
                style={
                  Object {
                    "width": "100%",
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#0086ff",
                      "borderRadius": 10,
                      "height": 56,
                      "justifyContent": "center",
                      "marginBottom": 15,
                      "opacity": 1,
                      "padding": 10,
                      "width": "100%",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "white",
                        "fontSize": 16,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    CARRINHO
                  </Text>
                </View>
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "white",
                      "borderColor": "#0086ff",
                      "borderRadius": 10,
                      "borderWidth": 1.5,
                      "height": 56,
                      "justifyContent": "center",
                      "opacity": 1,
                      "padding": 10,
                      "width": "100%",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#0086ff",
                        "fontSize": 16,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    CONTINUAR COMPRA
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </Modal>
    </View>
  </RCTSafeAreaView>,
  " ",
]
`;
